org.osgi.test.junit4/src/main/java/org/osgi/test/junit4/properties/PropertiesConverter.java

```diff
@@ -41,60 +42,19 @@ private static Object toValue(Description desc, Property entry) {
 
 		boolean primitive = entry.type()
 			.equals(Type.PrimitiveArray);
-		String[] value = getRawValue(desc, entry);
+		String[] value = getRawValue(desc, entry.value(), entry.source(), entry.type());
+
+		Object[] templateParams = Arrays.stream(entry.templateArguments())
+			.map(ta -> getRawValue(desc, ta.value(), ta.source(), Type.Scalar)[0])
+			.toArray();
 
 		Object result = createArray(entry.scalar(), primitive, value.length);
 		int i = 0;
 		for (String v : value) {
-			Object val = null;
-
-			if (v != null) {
-				switch (entry.scalar()) {
-					case Boolean :
-						Boolean booleanValue = Boolean.valueOf(v);
-						val = primitive ? booleanValue.booleanValue() : booleanValue;
-						break;
-
-					case Byte :
-						Byte byteVal = Byte.valueOf(v);
-						val = primitive ? byteVal.byteValue() : byteVal;
-						break;
-
-					case Character :
-						char charVal = v.charAt(0);
-						val = primitive ? charVal : Character.valueOf(charVal);
-						break;
-
-					case Double :
-						Double doubleVal = Double.valueOf(v);
-						val = primitive ? doubleVal.doubleValue() : doubleVal;
-						break;
-
-					case Float :
-						Float floatVal = Float.valueOf(v);
-						val = primitive ? floatVal.floatValue() : floatVal;
-						break;
-
-					case Integer :
-						Integer integerVal = Integer.valueOf(v);
-						val = primitive ? integerVal.intValue() : integerVal;
-						break;
-
-					case Long :
-						Long longVal = Long.valueOf(v);
-						val = primitive ? longVal.longValue() : longVal;
-						break;
-
-					case Short :
-						Short shortVal = Short.valueOf(v);
-						val = primitive ? shortVal.shortValue() : shortVal;
-						break;
-
-					case String :
-						val = v;
-						break;
-				}
+			if (v != null && templateParams.length > 0) {
+				v = String.format(v, templateParams);
```

Comment:
The use of format strings opens up a mess of user errors (e.g. using non-String consuming conversions). Maybe it would be safer to use simple placeholders like logging. e.g. `{}`.